src/document/DocumentManager.ts

```diff
@@ -116,6 +119,10 @@ export class DocumentManager {
 
   async saveFile() {
     const content = JSON.stringify(this.model.asSavedDocument(), undefined, 4);
+    if (!VERSIONS["v0.0.1"].validate(this.model.asSavedDocument())) {
+      console.error("Invalid Doc JSON: " + "\n" + JSON.stringify(documentSchema) + "\n" + content);
+      return;
+    }
```

Comment:
Is the purpose of this for developers to check that their documents are saved with the right schema? For example, if they messed something up in the document model, it would no longer be spec compliant, and this would let them know? If that's the case, I'd make this a warning instead of an error.

Reply:
It's most important for this to be in the open function in addition to save.